dht/dhtcore/NodeStore.c

```diff
@@ -1001,6 +1001,8 @@ static struct Node_Link* discoverLinkC(struct NodeStore_pvt* store,
     }
 
     struct Node_Two* parent = closest->child;
+    uint64_t realPathParentChild = pathParentChild == 1 ?
+        closest->cannonicalLabel : pathParentChild;
```

Comment:
Ahh ok now I remember what this is doing. Normally if pathParentChild == 1, parent == child.. because it is searching for the node in the tree and it found him. It is possible that pathParentChild == 1 and parent != child.. this is because the parent has restarted and the slots in it's switch were populated in a different order (or the child node disconnected and someone else connected), this is normal but complex to handle.

There is also a minor possibility that we are being lied to. The lie would be to modify a packet which is destine for us by changing the label in-flight so that we will store an invalid path. In the interest of being ~somewhat resilient to DoS, we try not to accept (unlikely) news without verifying it.

The intended behaviour when a packet comes in with a different switch slot is to add the parent to the rumorMill so that it will get a GetPeers request which will straiten out the issue (perhaps this is the bug...)

Inside of Janitor, the renumberMill is known as the externalMill because it is the only one which is influenceable from outside of Janitor...

It is called here:
https://github.com/cjdelisle/cjdns/blob/master/dht/dhtcore/Janitor.c#L617

To test, perhaps try adding this:
```c
RumorMill_addNode(store->renumberMill, &child->address);
// checking for DoS by way of a lying switch
if (discoveredPath == child->address.path || Node_getReach(child) == 0) {
    NodeStore_unlinkNodes(&store->pub, closest);
}
```

Right here:
https://github.com/cjdelisle/cjdns/blob/master/dht/dhtcore/NodeStore.c#L1051